assignment-1/Main.java

```
@@ -0,0 +1,624 @@
+        this.teams = teams;
+        matches = new ArrayList<>();
+        allBowlers = new ArrayList<>();
+        allAllRounders = new ArrayList<>();
+        allBatsmen = new ArrayList<>();
+        ArrayList<Player> allWicketKeepers = new ArrayList<>();
+        allPlayers = new ArrayList<>();
+
+        for (Player player : players) {
+            allPlayers.add(player);
+            switch (player.getRole()) {
+                case "BATSMAN" -> allBatsmen.add(player);
+                case "BOWLER" -> allBowlers.add(player);
+                case "WICKET KEEPER" -> allWicketKeepers.add(player);
+                default -> allAllRounders.add(player);
+            }
+        }
+
+        fixSchedule();
+        Processor.writeToCSV(matches);
+    }
+
+    public void fixSchedule() {
+        for (Team homeTeam : teams) {
+            for (Team awayTeam : teams) {
+                if (homeTeam.getTeamId() == awayTeam.getTeamId())
+                    continue;
+                Match match = new Match(homeTeam, awayTeam, homeTeam.getHomeGround());
+                matches.add(match);
+            }
+        }
+        Collections.shuffle(matches);
+        int identifier = 1;
+        for (Match match : matches) {
+            match.matchId = identifier;
+            identifier++;
+        }
+    }
+
+    public boolean isTeamPresent(String teamName) {
+        teamName = teamName.toLowerCase();
+        for (Team team : teams) {
+            if (team.getName().toLowerCase().equals(teamName)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public Team getTeam(String teamName) {
+        teamName = teamName.toLowerCase();
+        Team result = null;
+        for (Team team : teams) {
+            if (team.getName().toLowerCase().equals(teamName)) {
+                result = team;
+                break;
+            }
+        }
+        return result;
```

If the result is still null, you can probably throw a custom exception instead of returning null, which can be caught in the calling method.